web/angular-wallet/src/app/setup/setup.component.ts

```diff
@@ -1,54 +1,73 @@
-import { Component, OnInit } from '@angular/core';
-import { constants } from '../constants';
-import { I18nService } from '../layout/components/i18n/i18n.service';
-import { MatSelectChange } from '@angular/material';
-import { Account } from '@burstjs/core';
-import { StoreService } from 'app/store/store.service';
-import { environment } from 'environments/environment.hmr';
-import { FormControl } from '@angular/forms';
-import { ActivatedRoute } from '@angular/router';
-import { Settings } from 'app/settings';
+import {Component, OnInit} from '@angular/core';
+import {constants} from '../constants';
+import {I18nService} from '../layout/components/i18n/i18n.service';
+import {MatSelectChange} from '@angular/material';
+import {StoreService} from 'app/store/store.service';
+import {environment} from 'environments/environment.hmr';
+import {FormControl} from '@angular/forms';
+import {ActivatedRoute} from '@angular/router';
+import {Settings} from 'app/settings';
+
+interface NodeDescriptor {
+  url: string;
+  version: string;
+}
 
 @Component({
-    selector: 'app-setup',
-    styleUrls: ['./setup.component.scss'],
-    templateUrl: './setup.component.html'
+  selector: 'app-setup',
+  styleUrls: ['./setup.component.scss'],
+  templateUrl: './setup.component.html'
 })
 export class SetupComponent implements OnInit {
 
-    public selectedNode = new FormControl();
-    public languages: Array<any>;
-    public currentLanguage: any;
-    public settings: Settings;
-
-    public nodes = constants.nodes.map(({address, port}) => `${address}:${port}`)
-        .concat(environment.defaultNode);
-
-    constructor(private i18n: I18nService, private storeService: StoreService,
-        private route: ActivatedRoute) {
-    }
-
-    ngOnInit() {
-        this.languages = constants.languages;
-        this.currentLanguage = this.i18n.currentLanguage;
-        this.settings = this.route.snapshot.data.settings as Settings;
-        this.selectedNode.setValue(this.settings.node);
-        this.selectedNode.valueChanges.subscribe(this.setNode())
-    }
-
-    private setNode(): (value: string) => void {
-        return async (value) => {
-            const settings = await this.storeService.getSettings();
-            this.storeService.saveSettings({
-                ...settings,
-                node: value
-            });
-        };
-    }
-
-    setLanguage(event: MatSelectChange) {
-        this.currentLanguage = event.value;
-        this.i18n.setLanguage(event.value);
-    }
+  public selectedNode = new FormControl();
+  public languages: Array<any>;
+  public currentLanguage: any;
+  public settings: Settings;
+
+  public nodes = SetupComponent.createNodeList();
```

Comment:
This is a pattern I don't see every day. Typically I'd use `this` over referring to the class directly. Not sure if this is safe.

Reply:
As it does not use members of the object,  it can be static. Webstorm even suggest this in React apps. But could be also outside the class. Static methods are memory efficient as not part of the final instance and needs to be instantiated only once